src/ssc_joystick_ros.cpp

```diff
@@ -468,6 +470,9 @@ int main(int argc, char **argv)
 
     config_ok &= AS::readJsonWithError(mod_name, json_obj, "vel_controller_name", &vel_controller_name);
 
+    config_ok &= AS::readJsonWithError(mod_name, json_obj, "engage_speed_module", &engage_speed_module);
+    config_ok &= AS::readJsonWithError(mod_name, json_obj, "engage_steering_module", &engage_steering_module);
+
```

Comment:
This implicit conversion bothers me. We are reading the value `true` or `false` but storing it in an int. If we plan to use an int for storage, we should use an `if` to convert the `bool` from the JSON into the int value explicitly like this:

```
bool engage_speed_bool = false;
bool engage_steering_bool = false;
config_ok &= AS::readJsonWithError(mod_name, json_obj, "engage_speed_module", &engage_speed_bool);
config_ok &= AS::readJsonWithError(mod_name, json_obj, "engage_steering_module", &engage_steering_bool);
engage_speed_module = engage_speed_bool ? 1 : 0;
engage_steering_module = engage_steering_bool ? 1 : 0;
```

Reply:
Though my preference would be to store it internally as a bool and convert it to an int when the message is populated. Thoughts on this from @Daniel-Stanek ?